### PR TITLE
ACQ-101: Add custom tracking context for marketingPopupPrompt, anonSubscribeNow, and printBannerUsa

### PR DESCRIPTION
**Pull request by remybach**
[Ticket](https://financialtimes.atlassian.net/browse/ACQ-101)

### CODE REVIEW

File: client/utils.js [hunk 1]
```diff
@@ -37,16 +37,16 @@ const updateMessageEventCount = (messageId, event) => {
 };
 
 module.exports = {
-	generateMessageEvent: function ({ messageId, position, flag, variant }={}) {
+	generateMessageEvent: function ({ messageId, position, flag, trackingContext, variant }={}) {
```
Comment (umbobabo) [inline client/utils.js:40]: I would set a default value for trackingContext. 
```suggestion
	generateMessageEvent: function ({ messageId, position, flag, trackingContext = {}, variant }={}) {
```
or you could have an error when you do object.assign if that is not passed for other messages

Comment (remybach) [reply client/utils.js:40]: > `Object.assign()` does not throw on null or undefined sources.
> - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign

I did test this on other messages and they worked fine. Can't hurt to put a default in though so I'll update it.

Comment (umbobabo) [reply client/utils.js:40]: Don't change it then. I thought it was throwing on `undefined`

**Comment by umbobabo** [comment]
Approved @remybach, pls do not release now but do it on Monday (except in case of extreme emergency, top priority)

**Comment by remybach** [comment]
@umbobabo You must think me a mad man to release this late on a Friday 🤣

**Comment by umbobabo** [comment]
> @umbobabo You must think me a mad man to release this late on a Friday 🤣

I think of you as a wise man instead, but I also know that sometimes we got super-duper-urgent-top-priority-we-cannot-wait-till-monday requests around messaging. 😉

**Comment by ludrob** [comment]
8 km v thymus hi +hh+y 6u.                                    
   N.      6 bbb 5

**Pull request merged**